genimage.c

```diff
@@ -831,8 +831,17 @@ int main(int argc, char *argv[])
 			struct image *child;
 
 			if (!part->image) {
+				/* allow image-less partitions that are present
+				 * in the partition table
+				 */
 				if (part->in_partition_table)
 					continue;
+				/* allow image-less partitions that are not in
+				 * partition table and should be filled with
+				 * zeros (sparse = false)
+				 */
+				if (!part->in_partition_table && !part->sparse)
```

Comment:
`part->in_partition_table` is always false here. I think it make sense to merge the two cases and just document both in the comment. So:
```
if (part->in_partition_table || !part->sparse)
	continue;
```